### ISSUE TITLE
How to include more than one exchange ? 

### ISSUE THREAD

**Issue by mishaker** [freqtrade/freqtrade#1710]
It would be useful to include more than one exchange to the bot. then fetching data from multiple sources via DataProvider. 
Does it make sense to have a list of exchange instead of only one? 
So in freqtradebot we would have something like: 
``` python
for exchange_name in self.config.get('exchanges'):
   self.exchange[] += ExchangeResolver(exchange_name)
```

**Comment by hroff-1902** [comment]
Oh, no... (my IMHO)
Seems to be too complicated.

If you (or someone else) concerned about simultaneous changes to the common settings in case of separate instances running with separate configuration files, this can efficiently be resolved by utilizing multiple configuration files.

I.e., instance A can run with `-c common.json -c exchangeA.json`,
instance B correspondently can run with `-c common.json -c exchangeB.json` in the command line where common.json contains common settings while exchangeA.json and exchangeB.json -- contain settings, specific for exchangeA and exchangeB. Not even only the 'exchange' part of the settings, but any settings if they are specific for an exchange. This use case could (should) be described in the docs.

With newly added sd-notify support and an ability to run the bot under systemd watchdog, it's also possible to run multiple instances of the bot as systemd services with different service name (and configs)...

**Comment by mishaker** [comment]
Yeah but my intention is more about using multiple exchanges data in a
single strategy.
Let’s say if price goes up on exchange A then buy on exchange B. Note that
here exchange A is just informative and doesn’t have to have an account
attached to it

**Comment by hroff-1902** [comment]
Aha, arbitrage strategies... This can be interesting, agree...

**Comment by hroff-1902** [comment]
In this case (only informative pairs from different exchanges) this can be achieved by extending the syntax for informative pairs in the strategy. Something like "exchangeB:pairname" or "pairname@exchangeB"...

**Comment by xmatthias** [comment]
Thats a much wider change than just the informative pair interface.

Assume you wanna check ETH/BTC on binance and bittrex. We'd need to capture the exchange with the caching, store multiple instances of the exchange at each time, ...

on the other hand - nothing prevents a strategy from importing and initializing exchange (or Dataprovider) itself (as long as the base `__init__()` is called) - and then running a "fetch-data" from there - since all caching is in the exchange anyway.

That would keep the mess away from the core bot - which I think is already way too complicated the way it is.

**Comment by iuvbio** [comment]
I'm a big proponent of multi-exchange support, even for trading, but for the moment I'd agree with @xmatthias that it make things more complicated and there's probably other stuff to figure out first..

**Comment by sahilharidas** [comment]
Another similar reason as to why this would be useful is for getting data one from exchange while trading on another.

In my case, I need to use the luno exchange, which is ccxt supported but it doesn't provide timeframe data so it isn't compatible with freqtrade. So it would be nice to check the price on binance, for example, and then buy/sell on luno

**Comment by xmatthias** [comment]
what you'd do is assume that both exchanges trade at the same rate. that's sometimes the case, but it won't be the case always.
This does open up trading opportunities eventually - but i think for what you're trying to do, it's a high risk.

**Comment by hroff-1902** [comment]
to be honest, I'm also thinking of it for long time
(remember my comment for informative pairs: we could allow fetching them, either in the syntax of pairname, `"binance:ETHBTC"`, or as a third parameter in the informative pair tuples: `("ETHBTC", "5m", "binance")`; that was a suggestion from @mishaker an year ago, IIRC)

for this, we need to have an ability to define exchange settings for a list of exchanges in the config
only one can still be used as the trading exchange, others -- only for fetching ticker/market/ohlcv info, probably via the Dataprovider

**Comment by nathanrhoda** [comment]
Was there ever any strategy made on this front I am looking at doing something similar. Also does Freqtrade support Luno i see it mentioned earlier in the comments yet when looking on line on freqtrade site i dont see it listed?

**Comment by criadoperez** [comment]
I agree that multiple exchanges is the next step to go. @nathanrhoda , are you suggesting programing this feature? If so, I may be able to help.

**Comment by nathanrhoda** [comment]
@criadoperez  Im in just wrapping up a few side projects but lets discuss how we can approach this.

**Comment by xmatthias** [comment]
@criadoperez @nathanrhoda just to be clear what this request entails.

It's about using a pair from a different exchange as informative pair.
Trading will not be possible on multiple exchanges (the complexity this would introduce is not worth it and will make the code a lot less maintainable - so it's very unlikely a PR trying to introduce this would be accepted) - if you require that, please run 2 seperate bot instances.

**Comment by criadoperez** [comment]
Its then clear that adding multiple exchanges in the logic wont be accepted as part of the freqtrade project.
Me and another programmer created a software that did actually trade with multiple exchanges, but it needs a little update and it hasn't been touched in a couple of years. I'll see if I can upload it to github and recover it. It could maybe work in tandem with freqtrade? @nathanrhoda would you be interested in taking a look at it?

**Comment by xmatthias** [comment]
There's arbitrage bots around which do support that too - google is your friend.

I'd however please ask people to not abuse our issue-tracker to discuss about other topics - thanks.
